src/test/java/org/fogbowcloud/manager/core/AaControllerTest.java

```diff
@@ -48,112 +48,174 @@ public void setUp() {
+    public void testAuthorizeWhenUnauthorizedRequestExceptionOnInstanceType() throws UnauthorizedRequestException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any(), 
     			Mockito.any(InstanceType.class))).thenReturn(false);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any(), Mockito.any(InstanceType.class));
     }
     
+    //test case: Check if authenticate method throws Unauthenticated exception when the federation token is valid. 
     @Test (expected = UnauthorizedRequestException.class)
-    public void testAuthorizeWhenUnauthorizedRequestExceptionOnOrder() throws FogbowManagerException {
+    public void testAuthorizeWhenUnauthorizedRequestExceptionOnOrderType() throws FogbowManagerException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any(), 
     			Mockito.any(Order.class))).thenReturn(false);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any(), Mockito.any(Order.class));
     }
     
+    //test case: Check if authenticate method throws Unauthenticated exception when the federation token is valid. 
     @Test (expected = UnauthorizedRequestException.class)
     public void testAuthorizeWhenUnauthorizedRequestException() throws FogbowManagerException {
+    	//set up
     	Mockito.when(this.authorizationPluginMock.isAuthorized(
     			Mockito.any(), 
     			Mockito.any())).thenReturn(false);
+    	
+    	//exercise/verify
     	this.aaController.authorize(Mockito.any(), Mockito.any());
     }
     
+    //test case: Check if create token is returning a valid token properly.
     @Test
     public void testGetLocalToken() throws FogbowManagerException, UnexpectedException {
+    	//set up
     	Mockito.when(this.localUserCredentialsMapperPluginMock.getCredentials(Mockito.any())).thenReturn(new HashMap<String, String>());
     	String accessId = "accessId";
     	User user = new User("id", "name");
     	Date date = new Date();
     	Map <String, String> attributes = new HashMap<String, String>();
-    	Token token = new Token(accessId, user, date, attributes);
-    	Mockito.when(this.localIdentityPluginMock.createToken(Mockito.any())).thenReturn(token);
-    	Assert.assertEquals(token, this.aaController.getLocalToken(null));
+    	Token expectedToken = new Token(accessId, user, date, attributes);
+    	Mockito.when(this.localIdentityPluginMock.createToken(Mockito.any())).thenReturn(expectedToken);
+    	
+    	//exercise
+    	Token aaControllerToken = this.aaController.getLocalToken(null);
+    	
+    	//verify
+    	Assert.assertEquals(expectedToken, aaControllerToken);
     }
     
+    //test case: Check if getLocalToken is returning FogbowManagerException from LocalIdentityPlugin properly. 
```

Comment:
a better doc would say something like "... is properly forwarding the exception thrown by ..."

Reply:
Much better described, I'll use this template now, thank you